trackintel/model/trips.py

```diff
@@ -101,13 +145,70 @@ def temporal_tracking_quality(self, *args, **kwargs):
 
         See :func:`trackintel.analysis.tracking_quality.temporal_tracking_quality`.
         """
-        return ti.analysis.tracking_quality.temporal_tracking_quality(self._obj, *args, **kwargs)
+        return ti.analysis.tracking_quality.temporal_tracking_quality(self, *args, **kwargs)
 
-    def generate_tours(self, *args, **kwargs):
+    def generate_tours(self, **kwargs):
         """
         Generate tours based on trips (and optionally staypoint locations).
 
         See :func:`trackintel.preprocessing.trips.generate_tours`.
         """
-        assert len(args) == 0, "When calling 'generate_tours' via the accessor all arguments must be keyword arguments"
-        return ti.preprocessing.trips.generate_tours(trips=self._obj, **kwargs)
+        return ti.preprocessing.trips.generate_tours(trips=self, **kwargs)
+
+
+# added GeoDataFrame and DataFrame manually afterwards such that our methods always come first
+class TripsGeoDataFrame(TrackintelGeoDataFrameWithFallback, TripsDataFrame, gpd.GeoDataFrame, pd.DataFrame):
```

Comment:
why is it subclassing from both gpd.GeoDataFrame and pd.DataFrame? I guess it's just for readability, but I would only put one of them

Reply:
In addition to that, why do we need `gpd.GeoDataFrame` on top of `TrackintelGeoDataFrameWithFallback`? Because it is simpler to access the functions in `gpd.GeoDataFrame`?

Reply:
It is not for readability. I added them to avoid some interactions on inheritance.
For example the `to_postgis` method. If we would have
````python
````
If I wouldn't added `gpd.GeoDataFrame` after `TripsDataFrame` we would have the following linear inheritance order:
`````
TrackintelDataFrameWithFallback -> GeoDataFrame -> ... -> TripsDataFrame -> DataFrame -> ...
`````
Meaning we inherit the `to_postgis` method from the

Reply:
The `GeoDataFrame` is to avoid annoying inheritance problems, but the `DataFrame` was indeed not necessary and I have removed it.